src/containers/TransactionPage/TransactionPanel/ActionButtonsMaybe.js

```diff
@@ -15,22 +19,102 @@ const ActionButtonsMaybe = props => {
     secondaryButtonProps,
     isListingDeleted,
     isProvider,
+    estimatedLineItem,
+    onTransition,
+    transaction,
+    redirectToCheckoutPageWithInitialValues,
+    onUpdateTxDetails
   } = props;
 
+  const handleUpdateLineItems = async (values, listing, transitionName) => {
+    const {
+      bookingDates,
+      quantity: quantityRaw,
+      ...otherOrderData
+    } = values;
+
+    const bookingMaybe = bookingDates
+      ? {
+        bookingDates: {
+          bookingStart: bookingDates.startDate,
+          bookingEnd: bookingDates.endDate,
+        },
+      }
+      : {};
+    const quantity = Number.parseInt(quantityRaw, 10);
+    const quantityMaybe = Number.isInteger(quantity) ? { quantity } : {};
+
+    const orderData = {
+      ...bookingMaybe,
+      ...quantityMaybe,
+      ...otherOrderData
+    };
+    try {
+      const txId = await onUpdateTxDetails(listing, orderData, transitionName);
```

Comment:
Why we need to catch here if onUpdateTxDetails has already catch the error?

Reply:
Sorry, my bad. Yes, we don't need to. I wrote this base on the `onSendTxDetails` func and did not pay attention about this detail.